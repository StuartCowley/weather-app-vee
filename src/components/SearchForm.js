import PropTypes from "prop-types";
import React from "react";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText }) {
  const handleInputChange = (event) => setSearchText(event.target.value);
  return (
    <div className="search-form">
      <input
        type="text"
        id="search"
        name="q"
        placeholder="Enter city..."
        onChange={handleInputChange}
        value={searchText}
      />
      <button type="submit">Search</button>
    </div>
    // </form>
  );
}

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
};
