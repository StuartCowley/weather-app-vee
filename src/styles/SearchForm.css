.search-form {
  display: flex;
  justify-content: center;
  margin-top: 0.5rem;
  margin-bottom: 2rem;
}

.input-field {
  width: 8rem;
  font-size: 1rem;
  border: 1px solid #e5e4e2;
  display: inline-block;
  box-sizing: border-box;
}

.search-button {
  background-color: #ffffff;
  border: 1px solid #e5e4e2;
  box-sizing: border-box;
  color: #222222;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  font-weight: 300;
  margin: 0;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  touch-action: manipulation;
  transition: box-shadow 0.2s, -ms-transform 0.1s, -webkit-transform 0.1s,
    transform 0.1s;
  user-select: none;
  -webkit-user-select: none;
  width: auto;
}

.search-button:focus-visible {
  box-shadow: #222222 0 0 0 2px, rgba(255, 255, 255, 0.8) 0 0 0 4px;
  transition: box-shadow 0.2s;
}

.search-button:active {
  background-color: #f7f7f7;
  border-color: #000000;
  transform: scale(0.96);
}

.search-button:disabled {
  border-color: #dddddd;
  color: #dddddd;
  cursor: not-allowed;
  opacity: 1;
}

@media (min-width: 992px) {
  .search-form {
    display: block;
  }
}
