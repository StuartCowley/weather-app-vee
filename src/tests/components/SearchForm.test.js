import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Nottingham",
    setSearchText: () => {},
    onSubmit: () => {},
  };
  it("render correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the submit button correctly", () => {
    const { getByTestId } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );

    expect(getByTestId("button-text")).toHaveTextContent(/search/i);
  });
});
