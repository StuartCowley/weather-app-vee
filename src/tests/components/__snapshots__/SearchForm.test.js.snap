// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm render correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      placeholder="Enter city..."
      type="text"
      value="Nottingham"
    />
    <button
      data-testid="button-text"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
